### PR TITLE
54 ios add frequency chain

### PR DESCRIPTION
**Pull request by ryansheh**

### CODE REVIEW

File: DSNP-Wallet/Source/Common/Extension/Wallet/TransactionHistoryItem+Wallet.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import Foundation
+import CommonWallet
+import IrohaCrypto
+import BigInt
+import SubstrateSdk
```
Comment (BenJamminnn) [inline DSNP-Wallet/Source/Common/Extension/Wallet/TransactionHistoryItem+Wallet.swift]: Do we need all these imports for this extension?

File: DSNP-Wallet/Source/Common/ExtrinsicManager/ExtrinsicManager+Msa.swift [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+              let primaryPublicKeyData = primaryUser.publicKey?.rawData(),
+              let secondaryAccountId = secondaryUser.getAccountId() else {
+            throw ExtrinsicError.BadSetup
+        }
+        
+        let addKeyPayload = AddKeyPayloadArg(msaId: 1,
+                                             expiration: 5,
+                                             newPublicKey: secondaryAccountId)
+        
+        guard let keyPayloadData = try? JSONEncoder().encode(addKeyPayload) else { return }
```
Comment (BenJamminnn) [inline DSNP-Wallet/Source/Common/ExtrinsicManager/ExtrinsicManager+Msa.swift]: perhaps we should throw an error if the encoding fails

File: DSNP-Wallet/Source/Common/ExtrinsicManager/ExtrinsicManager.swift [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    
+    private func setExtrinsicService(with senderId: Data) throws {
+        let chainRegistry = ChainRegistryFactory.createDefaultRegistry() //MARK: Does this need to be global
+        
+        guard let chain = chainRegistry.getChain(for: FrequencyChain.shared.id),
+              let connection = chainRegistry.getConnection(for: FrequencyChain.shared.id),
+              let runtimeService = chainRegistry.getRuntimeProvider(for: FrequencyChain.shared.id) else {
+            throw ExtrinsicError.BadSetup
+        }
+        let operationManager = OperationManager(operationQueue: .main)
```
Comment (BenJamminnn) [inline DSNP-Wallet/Source/Common/ExtrinsicManager/ExtrinsicManager.swift]: I think we should define a background queue here for the extrinsic service to work with

File: DSNP-Wallet/Source/Common/ExtrinsicManager/FrequencyChain.swift [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+enum Currency: BigUInt {
+    case dollar = 100000000 //one token = 100000000 units
+    case cent = 1000000
+    case millicent = 1000
+    case unit = 1
+}
+
+struct FrequencyChain {
+    static let shared = FrequencyChain()
+    let id = "496e2f8a93bf4576317f998d01480f9068014b368856ec088a63e57071cd1d49"
```
Comment (BenJamminnn) [inline DSNP-Wallet/Source/Common/ExtrinsicManager/FrequencyChain.swift:21]: where does this id come from?

File: DSNP-Wallet/Source/Common/Network/Wallet/WalletNetworkOperationFactory+Protocol.swift [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+//import Foundation
+//import CommonWallet
+//import RobinHood
+//import xxHash_Swift
+//import SubstrateSdk
+//import IrohaCrypto
+//import Starscream
+//import BigInt
+//
+//enum WalletNetworkOperationFactoryError: Error {
```
Comment (BenJamminnn) [inline DSNP-Wallet/Source/Common/Network/Wallet/WalletNetworkOperationFactory+Protocol.swift]: We should delete this file before merging

File: DSNP-Wallet/Source/Common/Services/ExtrinsicService/ExtrinsicFeeProxy.swift [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    var delegate: ExtrinsicFeeProxyDelegate? { get set }
+
+    func estimateFee(
+        using service: ExtrinsicServiceProtocol,
+        reuseIdentifier: ExtrinsicFeeId,
+        setupBy closure: @escaping ExtrinsicBuilderClosure
+    )
+}
+
+//final class ExtrinsicFeeProxy {
```
Comment (BenJamminnn) [inline DSNP-Wallet/Source/Common/Services/ExtrinsicService/ExtrinsicFeeProxy.swift]: We should remove this class if we don't need it

File: DSNP-Wallet/Source/Common/Services/RemoteSubscription/AssetsUpdatingService.swift [hunk 7]
```diff
@@ -81,94 +81,94 @@ final class AssetsUpdatingService {
...
-            closure: nil,
-            assetBalanceUpdater: assetBalanceUpdater,
-            transactionSubscription: transactionSubscription
-        )
-
-        return maybeSubscriptionId.map { subscriptionId in
-            SubscriptionInfo(subscriptionId: subscriptionId, accountId: accountId, asset: asset)
-        }
-    }
+//    private func createSubscription(
```
Comment (BenJamminnn) [inline DSNP-Wallet/Source/Common/Services/RemoteSubscription/AssetsUpdatingService.swift:105]: Do we still need these stubbed functions?

File: DSNP-Wallet/Source/Common/Services/ServiceCoordinator.swift [hunk 8]
```diff
@@ -41,12 +41,12 @@ extension ServiceCoordinator: ServiceCoordinatorProtocol {
         setup(chainRegistry: chainRegistry)
 
         accountInfoService.setup()
-        assetsService.setup()
+//        assetsService.setup()
```
Comment (BenJamminnn) [inline DSNP-Wallet/Source/Common/Services/ServiceCoordinator.swift]: if `accountInfoService` is doing the setup and the throttling do we still need the `assetsService`?

File: DSNP-Wallet/Source/Common/Substrate/Types/CallCodingPath.swift [hunk 9]
```diff
@@ -1,11 +1,15 @@
 import Foundation
 
-struct CallCodingPath: Equatable, Codable {
+struct CallCodingPath: Hashable, Codable {
     let moduleName: String
     let callName: String
 }
 
 extension CallCodingPath {
+//    var isSubstrateOrEvmTransfer: Bool {
```
Comment (BenJamminnn) [inline DSNP-Wallet/Source/Common/Substrate/Types/CallCodingPath.swift]: we can delete this, I don't think ERC20 transfers are on the roadmap

**Review by BenJamminnn — APPROVED** [review]
I left some comments, but mostly cleanup

File: DSNP-Wallet/Source/Common/ExtrinsicManager/FrequencyChain.swift (hunk 4 above)
Comment (ryansheh) [reply DSNP-Wallet/Source/Common/ExtrinsicManager/FrequencyChain.swift:21]: The ID is the genesis hash for the chain.

**Review by BenJamminnn — APPROVED** [review]

**Pull request merged**